From 80e7af1cf2979accbf14e5676c9a3dc894e370c7 Mon Sep 17 00:00:00 2001
From: Michael Auchter <michael.auchter@ni.com>
Date: Mon, 30 Dec 2019 12:11:16 -0600
Subject: [PATCH 10/37] board: x410: update psu_init_gpl to 2019.1

Update with the psu_init_gpl for Rev B generated by Vivado 2019.1

Signed-off-by: Michael Auchter <michael.auchter@ni.com>
---
 board/ni/zynqmp/x410-rev2/psu_init_gpl.c | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/board/ni/zynqmp/x410-rev2/psu_init_gpl.c b/board/ni/zynqmp/x410-rev2/psu_init_gpl.c
index 87d600b912f..789e71558c2 100644
--- a/board/ni/zynqmp/x410-rev2/psu_init_gpl.c
+++ b/board/ni/zynqmp/x410-rev2/psu_init_gpl.c
@@ -58,6 +58,7 @@ static unsigned long psu_clock_init_data(void)
 	psu_mask_write(0xFF5E005C, 0x063F3F07U, 0x06010C00U);
 	psu_mask_write(0xFF5E0100, 0x013F3F07U, 0x01010600U);
 	psu_mask_write(0xFF5E0060, 0x023F3F07U, 0x02010600U);
+	psu_mask_write(0xFF5E004C, 0x023F3F07U, 0x02031900U);
 	psu_mask_write(0xFF5E0068, 0x013F3F07U, 0x01010500U);
 	psu_mask_write(0xFF5E006C, 0x013F3F07U, 0x01010402U);
 	psu_mask_write(0xFF18030C, 0x00000003U, 0x00000000U);
@@ -84,7 +85,6 @@ static unsigned long psu_clock_init_data(void)
 	psu_mask_write(0xFD1A0060, 0x03003F07U, 0x03000100U);
 	psu_mask_write(0xFD1A0068, 0x01003F07U, 0x01000200U);
 	psu_mask_write(0xFD1A0080, 0x00003F07U, 0x00000200U);
-	psu_mask_write(0xFD1A0084, 0x07003F07U, 0x07000203U);
 	psu_mask_write(0xFD1A00B8, 0x01003F07U, 0x01000203U);
 	psu_mask_write(0xFD1A00BC, 0x01003F07U, 0x01000203U);
 	psu_mask_write(0xFD1A00C0, 0x01003F07U, 0x01000202U);
@@ -226,7 +226,7 @@ static unsigned long psu_ddr_init_data(void)
 	psu_mask_write(0xFD080068, 0xFFFFFFFFU, 0x000E0000U);
 	psu_mask_write(0xFD080090, 0xFFFFFFFFU, 0x02A04161U);
 	psu_mask_write(0xFD0800C0, 0xFFFFFFFFU, 0x00000000U);
-	psu_mask_write(0xFD0800C4, 0xFFFFFFFFU, 0x000000DDU);
+	psu_mask_write(0xFD0800C4, 0xFFFFFFFFU, 0x000000E0U);
 	psu_mask_write(0xFD080100, 0xFFFFFFFFU, 0x0800040CU);
 	psu_mask_write(0xFD080110, 0xFFFFFFFFU, 0x08261009U);
 	psu_mask_write(0xFD080114, 0xFFFFFFFFU, 0x28240008U);
@@ -290,13 +290,13 @@ static unsigned long psu_ddr_init_data(void)
 	psu_mask_write(0xFD080900, 0xFFFFFFFFU, 0x40800604U);
 	psu_mask_write(0xFD080904, 0xFFFFFFFFU, 0x00007FFFU);
 	psu_mask_write(0xFD08090C, 0xFFFFFFFFU, 0x3F000008U);
-	psu_mask_write(0xFD080910, 0xFFFFFFFFU, 0x0E00B03CU);
+	psu_mask_write(0xFD080910, 0xFFFFFFFFU, 0x0E00B004U);
 	psu_mask_write(0xFD080914, 0xFFFFFFFFU, 0x09094F4FU);
 	psu_mask_write(0xFD080918, 0xFFFFFFFFU, 0x09092B2BU);
 	psu_mask_write(0xFD080A00, 0xFFFFFFFFU, 0x40800604U);
 	psu_mask_write(0xFD080A04, 0xFFFFFFFFU, 0x00007FFFU);
 	psu_mask_write(0xFD080A0C, 0xFFFFFFFFU, 0x3F000008U);
-	psu_mask_write(0xFD080A10, 0xFFFFFFFFU, 0x0E00B03CU);
+	psu_mask_write(0xFD080A10, 0xFFFFFFFFU, 0x0E00B004U);
 	psu_mask_write(0xFD080A14, 0xFFFFFFFFU, 0x09094F4FU);
 	psu_mask_write(0xFD080A18, 0xFFFFFFFFU, 0x09092B2BU);
 	psu_mask_write(0xFD080B00, 0xFFFFFFFFU, 0x40800604U);
@@ -359,7 +359,6 @@ static unsigned long psu_ddr_init_data(void)
 	psu_mask_write(0xFD08151C, 0xFFFFFFFFU, 0x01266300U);
 	psu_mask_write(0xFD08152C, 0xFFFFFFFFU, 0x00041800U);
 	psu_mask_write(0xFD081530, 0xFFFFFFFFU, 0x70400000U);
-	psu_mask_write(0xFD0817C4, 0xFFFFFFFFU, 0x000E0000U);
 	psu_mask_write(0xFD0817DC, 0xFFFFFFFFU, 0x012643C4U);
 
 	return 1;
@@ -455,22 +454,22 @@ static unsigned long psu_mio_init_data(void)
 	psu_mask_write(0xFF18020C, 0x00003FFFU, 0x00000FC0U);
 	psu_mask_write(0xFF180138, 0x03FFFFFFU, 0x03FFFFFFU);
 	psu_mask_write(0xFF18013C, 0x03FFFFFFU, 0x03FFFFFFU);
-	psu_mask_write(0xFF180140, 0x03FFFFFFU, 0x00000000U);
+	psu_mask_write(0xFF180140, 0x03FFFFFFU, 0x033FEF5EU);
 	psu_mask_write(0xFF180144, 0x03FFFFFFU, 0x03FFFFFFU);
 	psu_mask_write(0xFF180148, 0x03FFFFFFU, 0x03FFFFFFU);
-	psu_mask_write(0xFF18014C, 0x03FFFFFFU, 0x00000000U);
+	psu_mask_write(0xFF18014C, 0x03FFFFFFU, 0x03FFFFFFU);
 	psu_mask_write(0xFF180154, 0x03FFFFFFU, 0x03FFFFFFU);
 	psu_mask_write(0xFF180158, 0x03FFFFFFU, 0x03FFFFFFU);
-	psu_mask_write(0xFF18015C, 0x03FFFFFFU, 0x00000000U);
+	psu_mask_write(0xFF18015C, 0x03FFFFFFU, 0x03E7FDBFU);
 	psu_mask_write(0xFF180160, 0x03FFFFFFU, 0x03FFFFFFU);
 	psu_mask_write(0xFF180164, 0x03FFFFFFU, 0x03FFFFFFU);
-	psu_mask_write(0xFF180168, 0x03FFFFFFU, 0x00000000U);
+	psu_mask_write(0xFF180168, 0x03FFFFFFU, 0x03FFFE7FU);
 	psu_mask_write(0xFF180170, 0x03FFFFFFU, 0x03FFFFFFU);
 	psu_mask_write(0xFF180174, 0x03FFFFFFU, 0x03FFFFFFU);
-	psu_mask_write(0xFF180178, 0x03FFFFFFU, 0x00000000U);
+	psu_mask_write(0xFF180178, 0x03FFFFFFU, 0x02FC0FBFU);
 	psu_mask_write(0xFF18017C, 0x03FFFFFFU, 0x03FFFFFFU);
 	psu_mask_write(0xFF180180, 0x03FFFFFFU, 0x03FFFFFFU);
-	psu_mask_write(0xFF180184, 0x03FFFFFFU, 0x00000000U);
+	psu_mask_write(0xFF180184, 0x03FFFFFFU, 0x0303FFF4U);
 	psu_mask_write(0xFF180200, 0x0000000FU, 0x00000000U);
 
 	return 1;
@@ -479,13 +478,14 @@ static unsigned long psu_mio_init_data(void)
 static unsigned long psu_peripherals_pre_init_data(void)
 {
 	psu_mask_write(0xFF5E0108, 0x013F3F07U, 0x01012302U);
+	psu_mask_write(0xFF5E0238, 0x00000001U, 0x00000001U);
 
 	return 1;
 }
 
 static unsigned long psu_peripherals_init_data(void)
 {
-	psu_mask_write(0xFD1A0100, 0x0000007CU, 0x00000000U);
+	psu_mask_write(0xFD1A0100, 0x00000044U, 0x00000000U);
 	psu_mask_write(0xFF5E0238, 0x001A0000U, 0x00000000U);
 	psu_mask_write(0xFF5E023C, 0x0093C018U, 0x00000000U);
 	psu_mask_write(0xFF5E0230, 0x00000008U, 0x00000000U);
-- 
2.43.0

